Stratis.Bitcoin.Tests/DBreezeSingleThreadSessionTest.cs

```diff
@@ -0,0 +1,207 @@
+        [Fact]
+        public void NBitcoinDeserializeWithNotSupportedClassThrowsException()
+        {
+            Assert.Throws(typeof(NotSupportedException), () =>
+            {
+                string test = "Should throw exception.";
+
+                DBreezeSingleThreadSession.NBitcoinDeserialize(Encoding.UTF8.GetBytes(test), typeof(string));
+            });
+        }
+
+        [Fact]
+        public void DoStartsTransaction()
+        {            
+            using (var session = new DBreezeSingleThreadSession("TestThread", "TestData/DBreezeSingleThreadSession/DoStartsTransaction"))
+            {
+                session.Do(() =>
+                {
+                    Assert.NotNull(session.Transaction);
+                });
+            }
+        }
+
+        [Fact]
+        public void DoAbleToAccessExistingTransactionData()
+        {
+            var dir = "TestData/DBreezeSingleThreadSession/DoAbleToAccessExistingTransactionData";
+            if (Directory.Exists(dir))
+            {
+                Directory.Delete(dir, true);
+            }
+            uint256[] data = SetupTransactionData(dir);
+
+            using (var session = new DBreezeSingleThreadSession("TestThread", dir))
+            {
+                session.Do(() =>
+                {
+                    var data2 = new uint256[data.Length];
+                    var transaction = session.Transaction;
+                    int i = 0;
+                    foreach (var row in transaction.SelectForward<byte[], byte[]>("Table"))
+                    {
+                        data2[i++] = new uint256(row.Key, false);
+                    }
+
+                    Assert.True(data.SequenceEqual(data2));
+                });
+            }
+        }
+
+        [Fact]
+        public void DoWithTypePerformsTask()
+        {
+            using (var session = new DBreezeSingleThreadSession("TestThread", "TestData/DBreezeSingleThreadSession/DoWithTypePerformsTask"))
+            {
+                var task = session.Do<string>(() =>
+                {
+                    return "TaskResult";
+                   
+                });
+                task.Wait();
+
+                Assert.Equal("TaskResult", task.Result);
+            }
+        }
+
+        private static uint256[] SetupTransactionData(string folder)
+        {
+            using (var engine = new DBreezeEngine(folder))
+            {
+                var data = new[]
+                          {
+                        new uint256(3),
+                        new uint256(2),
+                        new uint256(5),
+                        new uint256(10),
+                    };
+
+                using (var tx = engine.GetTransaction())
+                {
+                    foreach (var d in data)
+                        tx.Insert("Table", d.ToBytes(false), d.ToBytes());
+                    tx.Commit();
+                }
+
+                return data;
+            }
+        }
+    }
+}
```

Comment:
Good job on DBreeze tests 
One of the DBreeze requirements is running in the same thread the db was initiated
I think we may need one more test to check the Do delegate always runs in the same thread.